Sources/core/objects/Human.cpp

```diff
@@ -82,6 +84,274 @@ bool Human::TryMove(Dir direct)
+                            }
+                        }
+                        else
+                        {
+                            pulled_object_ = 0;
+                        }
+                    }
+                    else
+                    {
+                        if (CanPass(pulled_object_->GetRoot()->GetPassable(direct), pulled_object_->passable_level) || CanPass(pulled_object_->GetRoot()->GetNeighbour(direct)->GetPassable(D_ALL), pulled_object_->passable_level)
+|| CanPass(pulled_object_->GetRoot()->GetNeighbour(direct)->GetPassable(helpers::revert_dir(direct)), pulled_object_->passable_level))
+                        {
+                            auto new_owner = pulled_object_->GetRoot()->GetNeighbour(direct);
+                            pulled_object_->GetRoot()->RemoveItem(pulled_object_->GetId());
+                            new_owner->AddItem(pulled_object_->GetId());
+                            if(!pulled_object_->TryMove(D_LEFT))
+                            {
+                                pulled_object_ = 0;
+                            }
+                        }
+                        else if (CanPass(pulled_object_->GetRoot()->GetPassable(D_LEFT), pulled_object_->passable_level) || CanPass(pulled_object_->GetRoot()->GetNeighbour(D_LEFT)->GetPassable(D_ALL), pulled_object_->passable_level)
+|| CanPass(pulled_object_->GetRoot()->GetNeighbour(D_LEFT)->GetPassable(helpers::revert_dir(D_LEFT)), pulled_object_->passable_level))
+                        {
+                            auto new_owner = pulled_object_->GetRoot()->GetNeighbour(D_LEFT);
+                            pulled_object_->GetRoot()->RemoveItem(pulled_object_->GetId());
+                            new_owner->AddItem(pulled_object_->GetId());
+                            if(!pulled_object_->TryMove(direct))
+                            {
+                                pulled_object_ = 0;
+                            }
+                        }
+                        else
+                        {
+                            pulled_object_ = 0;
+                        }
+                    }
+                    break;
+                    case D_LEFT:
+                    if(y > 0)
+                    {
+                        if (CanPass(pulled_object_->GetRoot()->GetPassable(direct), pulled_object_->passable_level) || CanPass(pulled_object_->GetRoot()->GetNeighbour(direct)->GetPassable(D_ALL), pulled_object_->passable_level)
+|| CanPass(pulled_object_->GetRoot()->GetNeighbour(direct)->GetPassable(helpers::revert_dir(direct)), pulled_object_->passable_level))
+                        {
+                            auto new_owner = pulled_object_->GetRoot()->GetNeighbour(direct);
+                            pulled_object_->GetRoot()->RemoveItem(pulled_object_->GetId());
+                            new_owner->AddItem(pulled_object_->GetId());;
+                            if(!pulled_object_->TryMove(D_DOWN))
```

Comment:
Also, you are using a lot of copy-paste. Here you have 4 practically same pieces of the code - it would be logically to make them as separate function or something. (But for now it should be needed, because I suppose this code is not needed at all - because as I said all work should do `TryMove` function).